### ISSUE TITLE
Fatal errors make test run forever

### ISSUE THREAD

**Issue by ivan-klass** [etorreborre/specs2#822]
While using `org.specs2.mutable.SpecificationWithJUnit` (didn't checking other) fatal errors in tests (like missing jar dependency / incorrect jar version in classpath) make specification run forever, quick debug digging results in a loop within

[org.specs2.execute.ResultExecution#execute](https://github.com/etorreborre/specs2/blob/1acaab5aeeb5a5b0fc307bdcd9e846e690387cf4/common/shared/src/main/scala/org/specs2/execute/ResultExecution.scala#L21) -> 
[org.specs2.execute.ResultExecution#handleExceptionsPurely](https://github.com/etorreborre/specs2/blob/1acaab5aeeb5a5b0fc307bdcd9e846e690387cf4/common/shared/src/main/scala/org/specs2/execute/ResultExecution.scala#L26)

A very short example that makes test run forever is
```scala
import org.specs2.mutable.SpecificationWithJUnit

class Spec2ThrowableBug extends SpecificationWithJUnit {
  sequential

  def boom: String = throw new java.lang.NoSuchMethodError("Boom")

  "fatal stucks forever" >> {
    boom must beEqualTo("")
  }
}
```

**Comment by etorreborre** [comment]
I think this is fixing the issue. Let's wait on CI to see if the build is green.

**Comment by ivan-klass** [comment]
@etorreborre thank you very much, would you mind releasing a minor version with that fix?

**Comment by etorreborre** [comment]
I just released `4.9.3` with the fix: https://repo1.maven.org/maven2/org/specs2/specs2-core_2.13/4.9.3/
